src/NServiceBus.Transport.RabbitMQ.TransportTests/ConfigureRabbitMQTransportInfrastructure.cs

```diff
@@ -88,14 +82,32 @@ ConnectionFactory CreateConnectionFactory()
     {
         var connectionFactory = new ConnectionFactory
         {
-            UserName = connectionStringBuilder["username"].ToString(),
-            Password = connectionStringBuilder["password"].ToString(),
-            VirtualHost = connectionStringBuilder["virtualhost"].ToString(),
-            HostName = connectionStringBuilder["host"].ToString(),
             AutomaticRecoveryEnabled = true,
             UseBackgroundThreadsForIO = true
         };
 
+        object value;
```

Comment:
The code here can use some of the same suggested changes as the acceptance tests, but we've got another problem here. Unlike the acceptance tests, the `Cleanup` method in the transport tests appears to be still be called even if an exception is thrown in the `Configure` method. That means that `connectionStringBuilder` can be null. We also can't guarantee we have a fully formed connection string either, since this code still runs even when `ConnectionConfiguration.Create` fails to validate it.

We should probably look into making an upstream change in the transport tests to change this behavior, but in the mean time, we should add something to `Configure` to make sure it ran to completion that we can check in `Cleanup`.